### PR TITLE
Fix SwiftPackageManager dependencies mapping when it supports Mac Catalyst

### PR DESCRIPTION
**Pull request by danyf90**
Resolves https://github.com/tuist/tuist/issues/4186

Map maccatalyst to iOS with mac

### CODE REVIEW

File: Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift [hunk 1]
```diff
@@ -493,10 +493,11 @@ extension ProjectDescription.DeploymentTarget {
         package: [PackageInfo.Platform],
         packageName _: String
     ) throws -> Self {
-        if let packagePlatform = package.first(where: { $0.platformName == platform.rawValue }) {
+        if let packagePlatform = package.first(where: { $0.tuistPlatformName == platform.rawValue }) {
             switch platform {
             case .iOS:
-                return .iOS(targetVersion: packagePlatform.version, devices: [.iphone, .ipad])
+              let hasMacCatalyst = package.contains(where: { $0.platformName == "maccatalyst" })
+              return .iOS(targetVersion: packagePlatform.version, devices: hasMacCatalyst ? [.iphone, .ipad, .mac] : [.iphone, .ipad])
```
Comment (fortmarek) [inline Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift:500]: is adding `.mac` enough to support maccatalyst?

**Review by fortmarek — APPROVED** [review]
I wonder if we want to edit a fixture to cover this use-case. Wdyt?

File: Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift (hunk 1 above)
Comment (danyf90) [reply Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift]: AFAIK that's exactly what mac is supposed to do

**Comment by danyf90** [comment]
> I wonder if we want to edit a fixture to cover this use-case. Wdyt?

It would require a whole new dependency fixture to be set up as the current one builds for iOS so it wouldn't really test it. Given it's not a major use case I don't think it's worth at the moment and we can live with the unit tests

**Comment by fortmarek** [comment]
> It would require a whole new dependency fixture to be set up as the current one builds for iOS so it wouldn't really test it. Given it's not a major use case I don't think it's worth at the moment and we can live with the unit tests

Fair, we can rethink this in the future 🙂

File: Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift (hunk 1 above)
Comment (fortmarek) [reply Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift]: Cool! Never really used maccatylyst, so was not completely sure 😅

**Pull request merged**